server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -139,6 +142,9 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
                          compactionTaskQuery.getGranularitySpec().getSegmentGranularity(),
                          configuredSegmentGranularity);
                 indexingServiceClient.cancelTask(status.getId());
+
+                // Remove this from the locked intervals
+                taskToLockedIntervals.remove(status.getId());
```

Comment:
As mentioned earlier, this is not really needed. This would be the same as taskToLockedIntervals not containing lock for compaction task in the first place.

Reply:
Removed.